### DISCUSSION TITLE
Использование "Reverse Image Search Engine" как решение избежать дубликаты при загрузке и публикации

### DISCUSSION THREAD

**Discussion by kabalin** [PastVu/pastvu#473] [Ideas / Идеи и предложения]
Идея использования AI механизма анализа изображения могла бы помочь отсекать дубликаты и копии низкого качества уже на этапе загрузки и модерации, демонстрируя пользователю индентичные снимки с предложением убедиться, что загружаемый снимок уникален. Как сторонний эффект, механизм мог бы помочь определять местположения снимков без локации сравнивая их с базой существующих.

Задача, разумеется, не первоочередная и ресурсоемкая.

Описание принципа:
https://www.oreilly.com/library/view/practical-deep-learning/9781492034858/ch04.html

**Comment by paul-k-pastvu** [comment]
Да, эта функция давно напрашивается. 
Вероятно, нужно это включить в пул фич, связанных с редизайном модуля загрузки - предусмотреть в рамках тамошней логики какую-то библиотеку подключаемых внешних алгоритмов, для разных целей:
1) дубликаты/варианты
2) ориентация/поворот
3) удаление растра
4) подбор названий фото из внешних источников
5) ...

**Comment by paul-k-pastvu** [comment]
Кстати, это же уже начинали обсуждать в  #335
